checks/raw/branch_protection.go

```diff
@@ -72,8 +74,15 @@ func BranchProtection(c clients.RepoClient) (checker.BranchProtectionsData, erro
 		}
 
 		// TODO: if this is a sha, get the associated branch. for now, ignore.
-		if commit.MatchString(release.TargetCommitish) {
-			continue
+		switch c.(type) {
+		case *mockrepo.MockRepoClient:
+			if commit.MatchString(release.TargetCommitish) {
+				continue
```

Comment:
If we create a code path specifically for tests, the production code path will not get tested :) 

Update the unit test such that `MockRepo.URI()` returns appropriate value and remove this `switch` case here.